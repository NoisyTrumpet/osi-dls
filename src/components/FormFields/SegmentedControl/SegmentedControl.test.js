import React from 'react';
import { render } from '@testing-library/react';
import { Formik, Form } from 'formik';
import { SegmentedControl } from './';

const segments = [
  { name: 'apple', value: 'brands-apple-bold' },
  { name: 'google', value: 'brands-google-bold' },
  { name: 'calendar', value: 'calendar-calendar-bold' },
  { name: 'card', value: 'card-front-bold' }
];
const defaultSelected = 2;

describe('<Segmented Control />', () => {
  test('renders', () => {
    const { container } = render(
    <Formik>
      <Form>
        <SegmentedControl name="example" segments={segments} defaultSelected={defaultSelected} />
      </Form>
    </Formik>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
