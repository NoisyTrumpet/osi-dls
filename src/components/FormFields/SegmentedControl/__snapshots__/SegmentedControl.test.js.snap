// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Segmented Control /> renders 1`] = `
<form
  action="#"
>
  <div
    class="segmented-control-holder selected-2"
    test-id="segmented-control-test-id"
  >
    <div
      class="input-fieldset fieldset-container no-helpertext"
    >
      <fieldset
        name="example"
      >
        <legend
          class="screen-reader-text"
        />
        <ul
          class="segmented-control"
        >
          <li
            class="segmented-control-item"
          >
            <input
              class="input-field"
              id="brands-apple-bold"
              name="example"
              placeholder=""
              type="radio"
              value="brands-apple-bold"
            />
            <label
              class="segmented-control-label"
              for="brands-apple-bold"
            >
              <span
                class="typography screen-reader-text"
              />
              <span
                aria-hidden="true"
                class="icon icon-undefined icon-24"
                data-testid="icon"
              />
            </label>
          </li>
          <li
            class="segmented-control-item"
          >
            <input
              class="input-field"
              id="brands-google-bold"
              name="example"
              placeholder=""
              type="radio"
              value="brands-google-bold"
            />
            <label
              class="segmented-control-label"
              for="brands-google-bold"
            >
              <span
                class="typography screen-reader-text"
              />
              <span
                aria-hidden="true"
                class="icon icon-undefined icon-24"
                data-testid="icon"
              />
            </label>
          </li>
          <li
            class="segmented-control-item"
          >
            <input
              class="input-field"
              id="calendar-calendar-bold"
              name="example"
              placeholder=""
              type="radio"
              value="calendar-calendar-bold"
            />
            <label
              class="segmented-control-label"
              for="calendar-calendar-bold"
            >
              <span
                class="typography screen-reader-text"
              />
              <span
                aria-hidden="true"
                class="icon icon-undefined icon-24"
                data-testid="icon"
              />
            </label>
          </li>
          <li
            class="segmented-control-item"
          >
            <input
              class="input-field"
              id="card-front-bold"
              name="example"
              placeholder=""
              type="radio"
              value="card-front-bold"
            />
            <label
              class="segmented-control-label"
              for="card-front-bold"
            >
              <span
                class="typography screen-reader-text"
              />
              <span
                aria-hidden="true"
                class="icon icon-undefined icon-24"
                data-testid="icon"
              />
            </label>
          </li>
        </ul>
      </fieldset>
    </div>
  </div>
</form>
`;
